### PR TITLE
signal errors while getting file from JAR using optional

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: src/java_bytecode/jar_file.cpp [hunk 1]
```diff
@@ -96,21 +98,23 @@ static std::string trim(
...
-    {
-      const auto key_end=std::find(line.cbegin(), line.cend(), ':');
-      if(key_end!=line.cend())
-        out.emplace(
-          trim(line.cbegin(), key_end),
-          trim(std::next(key_end), line.cend()));
-    }
+    const auto key_end=std::find(line.cbegin(), line.cend(), ':');
+    if(key_end!=line.cend())
+      out.emplace(
```
Comment (thk123) [inline src/java_bytecode/jar_file.cpp]: Nit: body spread over multiple lines so use braces

File: src/java_bytecode/java_class_loader.cpp [hunk 2]
```diff
@@ -91,11 +91,14 @@ optionalt<java_bytecode_parse_treet> java_class_loadert::get_class_from_jar(
   debug()
     << "Getting class `" << class_name << "' from JAR " << jar_file << eom;
 
-  std::string data =
+  auto data =
     jar_pool(class_loader_limit, jar_file).get_entry(jar_index_it->second);
 
+  if(!data.has_value())
+    return {};
```
Comment (thk123) [inline src/java_bytecode/java_class_loader.cpp:98]: Previous behaviour here was to terminate the program and now returns an empty parse tree. This seems reasonable but I wonder if this now hides typos?

**Review by thk123 — COMMENTED** [review]
Consider whether the non-fatal nature of `get_class_from_jar` is what we want or whether a caller should be terminating the program. 

Other than that LGTM

File: src/java_bytecode/java_class_loader.cpp (hunk 2 above)
Comment (kroening) [reply src/java_bytecode/java_class_loader.cpp:98]: This isn't an empty parse tree, but a nullopt one -- very different from an empty string!

Note that the sole user checks existence of the file before calling, i.e., this should be unreachable.

**Comment by thk123** [comment]
Approved pending TG bump: diffblue/test-gen#1784

**Review by thk123 — APPROVED** [review]
TG bump passes

**Pull request merged**